chia/wallet/wallet_node.py

```diff
@@ -297,9 +305,10 @@ def _close(self):
 
     async def _await_closed(self, shutting_down: bool = True):
         self.log.info("self._await_closed")
-
         if self._server is not None:
             await self.server.close_all_connections()
+        if self._weight_proof_handler is not None:
+            self._weight_proof_handler.cancel_weight_proof_tasks()
```

Comment:
this doesn't look right. It looks like this should be in `_close()` and here you should `await` all tasks having been terminated.

Reply:
this was moved from wallet_state_manager where it was in the await_closed
i moved it now but looking at the code am not sure whats the purpose of the separation between await_closed and close